sympy/solvers/solveset.py

```diff
@@ -340,10 +340,26 @@ def _invert_complex(f, g_ys, symbol):
                                imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, exp) or (f.is_Pow and f.base == S.Exp1):
-        if isinstance(g_ys, FiniteSet):
+        if isinstance(g_ys, ImageSet):
+            # can solve upto `(d*exp(exp(a*x**n + b*x**(n-1) + ... + f)) + c)` format.
+            # Further can be improved to `(d*exp(exp(...(exp(a*x**n + b*x**(n-1) + ... + f))...) + c)`.
+            g_ys_expr = g_ys.lamda.expr
+            g_ys_vars = g_ys.lamda.variables
+            soln = solvify(f.exp, symbol, S.Complexes)
```

Comment:
Since `solvify` uses `solveset` which uses `invert` this is somewhat circular. I don't think that `solvify` should be used here. Also `solvify` returns incomplete solutions.

Shouldn't this just use `_invert_complex`?

Reply:
This actually solves a different equation.
i.e., 

`invert` try to solve -- >` d*exp(exp(exp(a*x**n + b*x**(n-1) + ... + f))) + c`
Whereas `solvify` tries to solve --> `a*x**n + b*x**(n-1) + ... + f)`

Reply:
What about using the `solve` function?
Or any other suggestions for such a situation?

Reply:
I don't think that any of `solve`, `solvify` or `solveset` should be used here as this is a lower level routine.

Why can't this be handled by `invert` like the other cases here?

Reply:
> I don't think that any of solve, solvify or solveset should be used here as this is a lower level routine.
> 
> Why can't this be handled by invert like the other cases here?

It could be handled but it can't be that smooth as `invert` can't solve equations like `a*x**n + b*x**(n-1) + ... + f`.

Reply:
Is that why `solveset` can't solve this:
```
In [21]: solveset(sin(x**2 - 1), x)
Out[21]: 
⎧  │         ⎛   ⎛ 2    ⎞    ⎞⎫
⎨x │ x ∊ ℂ ∧ ⎝sin⎝x  - 1⎠ = 0⎠⎬
⎩  │                          ⎭
```

Reply:
yep, I think this is pretty much a similar situation.